src/server/pfs/server/driver_fsck.go

```diff
@@ -134,163 +134,223 @@ func (e ErrMissingBranchHead) Error() string {
 	return fmt.Sprintf("consistency error: branch %s does not have a head commit", e.Branch)
 }
 
-// fsck verifies that pfs satisfies the following invariants:
-// 1. Branch provenance is transitive
-// 2. Head commit provenance has heads of branch's branch provenance
-// If fix is true it will attempt to fix as many of these issues as it can.
-func (d *driver) fsck(ctx context.Context, fix bool, cb func(*pfs.FsckResponse) error) error {
-	onError := func(err error) error { return cb(&pfs.FsckResponse{Error: err.Error()}) }
+func checkBranchProvenances(bi *pfs.BranchInfo, branchInfos map[string]*pfs.BranchInfo, onError func(error) error) error {
+	// we expect the branch's provenance to equal the union of the provenances of the branch's direct provenances
+	// i.e. union(branch, branch.Provenance) = union(branch, branch.DirectProvenance, branch.DirectProvenance.Provenance)
+	direct := bi.DirectProvenance
+	union := []*pfs.Branch{bi.Branch}
```

Comment:
It is a bit strange to call this a union. It looks like the union comes into play with the equal check, which also seems a bit strange that ordering doesn't matter there.

Reply:
union sounds fine to me but lemme know if you have a better description

Reply:
It seems strange because it doesn't remove duplicates. The duplicates are handled by the equality check. Just calling this provenance seems more clear to me.

Reply:
how about `cumulativeProvenance` instead of `union`?

Reply:
provenance refers to the transitive closure already, so I don't think the word cumulative adds information. That is why we specify direct provenance to indicate just the immediately provenant branches / commits.

Reply:
We're comparing two sets of provenances - one is simple and nameless - branch provenance + branch itself - do you want to call the other transitiveProvenance? Or stick with union?

Reply:
Going with `followedProvenances`

Reply:
That is fine.